### PR TITLE
update to use ce-arquillian-cube

### PR DESCRIPTION
**Pull request by rcernich**
Modified JDG testsuite to use new ce-arquillian-cube integration.

### CODE REVIEW

File: jdg/src/test/java/org/jboss/test/arquillian/ce/jdg/JdgTest.java [hunk 1]
```diff
@@ -65,10 +60,9 @@
                 @TemplateParameter(name = "HOSTNAME_HTTPS", value="jdg-http-route.openshift"),
                 @TemplateParameter(name = "HTTPS_NAME", value="jboss"),
                 @TemplateParameter(name = "HTTPS_PASSWORD", value="mykeystorepass"),
-                @TemplateParameter(name = "IMAGE_STREAM_NAMESPACE", value="${kubernetes.namespace}")})
+                @TemplateParameter(name = "IMAGE_STREAM_NAMESPACE", value="openshift")})
```
Comment (alesj) [inline jdg/src/test/java/org/jboss/test/arquillian/ce/jdg/JdgTest.java]: Why the hardcoding?
Just grab if from config ... ?

File: jdg/src/test/resources/arquillian.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    </extension>
+  
+    <!-- Example configuration for arquillian openshift extension -->
+    <extension qualifier="openshift">
+        <!-- Properties describing the location of the OpenShift instance.
+        <property name="originServer">https://your-ose-instance.com:8443</property>
+        <property name="namespace">test-namespace</property>
+        -->
+        <!-- The file containing the pod definition. Used to instantiate the JBoss instance. -->
+        <property name="definitionsFile">src/test/resources/testrunner-pod.json</property>
```
Comment (alesj) [inline jdg/src/test/resources/arquillian.xml]: Hmmm, should this really point to the source?
Imo this needs to be referenced from where it's gonna be after compile, etc

File: jdg/src/test/java/org/jboss/test/arquillian/ce/jdg/JdgTest.java (hunk 1 above)
Comment (rcernich) [inline jdg/src/test/java/org/jboss/test/arquillian/ce/jdg/JdgTest.java]: This removes the need of using a custom image stream resource.  I recommend installing the image stream in another project and referencing it through the namespace parameter.  Parameter values can be overridden using system variables, so no issue here.  To point to an image stream in another project: -DIMAGE_STREAM_NAMESPACE=myproject

File: jdg/src/test/resources/arquillian.xml (hunk 2 above)
Comment (rcernich) [inline jdg/src/test/resources/arquillian.xml]: No.  arquillian-cube doesn't really do any processing.  The test container is described simply as a pod.  This resource is pretty much the same as what you would have gotten from wildfly or eap spi.  All it's used for is to run tests within openshift, i.e. this is the @RunInPodDeployment container.

File: jdg/src/test/java/org/jboss/test/arquillian/ce/jdg/JdgTest.java (hunk 1 above)
Comment (alesj) [inline jdg/src/test/java/org/jboss/test/arquillian/ce/jdg/JdgTest.java]: Ah, ok, yes, true:

                addParameterValues(values, readParameters(template, configuration, resolver), false);
                addParameterValues(values, System.getenv(), true);
                addParameterValues(values, System.getProperties(), true);

File: jdg/src/test/resources/arquillian.xml (hunk 2 above)
Comment (alesj) [inline jdg/src/test/resources/arquillian.xml]: Hmmm, sure, but imo, the source is not what you should be referencing ever.
e.g. the directory schema can change -- from Maven-like to somethingnew ...
It's the classpath that's always gonna be the same.

Comment (rcernich) [inline jdg/src/test/resources/arquillian.xml]: Possibly, but that's how arquillian-cube currently works.  Also, if the directory structure changes, then you'd need to change this setting as well.  I don't think it's overly harmful that this is statically defined.  If you want to replace it, you can always use a different file and specify that through -Darq.extension.openshift.definitionsFile=my-custom-testrunner-pod.json.

**Pull request closed without merging**